backends/platform/ios7/ios7_osys_video.mm

```diff
@@ -140,9 +144,19 @@ static inline void execute_on_main_thread(void (^block)(void)) {
 	// Create the screen texture right here. We need to do this here, since
 	// when a game requests hi-color mode, we actually set the framebuffer
 	// to the texture buffer to avoid an additional copy step.
-	execute_on_main_thread(^ {
-		[[iOS7AppDelegate iPhoneView] createScreenTexture];
-	});
+	// Free any previous screen texture and create new to handle format changes
+	_videoContext->screenTexture.free();
+
+	if (format && *format == Graphics::PixelFormat(4, 8, 8, 8, 8, 0, 8, 16, 24)) {
+		// ABGR8888 (big endian)
+		_videoContext->screenTexture.create((uint16) getSizeNextPOT(_videoContext->screenWidth), (uint16) getSizeNextPOT(_videoContext->screenHeight), Graphics::PixelFormat(4, 8, 8, 8, 8, 0, 8, 16, 24));
+	} else if (format && *format == Graphics::PixelFormat(4, 8, 8, 8, 8, 24, 16, 8, 0)) {
+		// RGBA8888 (big endian)
+		_videoContext->screenTexture.create((uint16) getSizeNextPOT(_videoContext->screenWidth), (uint16) getSizeNextPOT(_videoContext->screenHeight), Graphics::PixelFormat(4, 8, 8, 8, 8, 24, 16, 8, 0));
+	} else {
+		// Assume RGB565
+		_videoContext->screenTexture.create((uint16) getSizeNextPOT(_videoContext->screenWidth), (uint16) getSizeNextPOT(_videoContext->screenHeight), Graphics::PixelFormat(2, 5, 6, 5, 0, 11, 5, 0, 0));
+	}
```

Comment:
Since you are using `kEAGLColorFormatRGBA8` for the internal GLES buffer color format now, maybe it would also make sense to use ABGR8888 as default instead of RGB565?

So basically:
- If format is RGB565, use that format
- Else if format is RGBA8888, use that format
- Else use ABGR8888

This would only impact 8bit color games (CLUT8) I think, but the benefit would be:
- avoid the additional color conversion from the screen texture to the internal GLES buffer (although it is probably hardware accelerated, so it would only be marginal gain)
- avoid losing precision on the colors (the palette is defined using 8 bit per color channel).

However that change is not trivial and involves additional changes (see for example `OSystem_iOS7::_gamePalette` and `OSystem_iOS7::drawDirtyRect` that currently both assume the texture color format is 16 bits for CLUT8). So maybe it could be kept for a separate pull request.